### PR TITLE
[gql_http_link] Add response headers to context

### PR DESCRIPTION
**Pull request by agent3bood**
Related issue https://github.com/gql-dart/gql/issues/119

### CODE REVIEW

File: links/gql_http_link/lib/src/link.dart [hunk 1]
```diff
@@ -32,13 +32,21 @@ class HttpLinkResponseContext extends ContextEntry {
...
 
+  /// HTTP response headers
+  final Map<String, String> headers;
+
   const HttpLinkResponseContext({
     @required this.statusCode,
-  }) : assert(statusCode != null);
+    @required this.headers,
+  }) : assert(
+          statusCode != null,
```
Comment (joshuadeguzman) [inline links/gql_http_link/lib/src/link.dart:42]: Thanks for this. Maybe we can add some explicit error messages here?

File: links/gql_http_link/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 0.2.10
+
+- add `headers` to `HttpLinkResponseContext`
```
Comment (joshuadeguzman) [inline links/gql_http_link/CHANGELOG.md:3]: I still think that this should be included on a release commit.

Comment (agent3bood) [reply links/gql_http_link/CHANGELOG.md:3]: What do you mean, should I remove this change?

File: links/gql_http_link/lib/src/link.dart (hunk 1 above)
Comment (agent3bood) [reply links/gql_http_link/lib/src/link.dart]: I have followed the same assert style on the project, split it in two lines and keep it without error message.

File: links/gql_http_link/CHANGELOG.md (hunk 2 above)
Comment (micimize) [reply links/gql_http_link/CHANGELOG.md:3]: My understanding is we [update the changelog in the change PR](https://github.com/gql-dart/gql/pull/114#issuecomment-640757142), so this is right. @klavs seems we should probably make a more detailed contributor doc.

File: links/gql_http_link/lib/src/link.dart (hunk 1 above)
Comment (micimize) [reply links/gql_http_link/lib/src/link.dart]: Right now this checks `statusCode != null` with the message `headers != null` should be:
```dart
assert(statusCode != null),
assert(headers != null);
```
The messages here would be too generic to be worth including IMO

Comment (agent3bood) [reply links/gql_http_link/lib/src/link.dart]: I fixed this mistake.

**Comment by agent3bood** [comment]
@micimize What should I do now? 
I am not sure if I am required to do something or if we are waiting for @klavs

**Comment by micimize** [comment]
@agent3bood it LGTM so yeah it's just waiting on @klavs

**Comment by klavs** [comment]
> My understanding is we update the changelog in the change PR, so this is right. @klavs seems we should probably make a more detailed contributor doc.

@micimize, agreed. I'm not in favor of the Angular-style commit messages. That's been a major blocker for me offering small fixes to projects like graphql_flutter. I never liked spending more time reading the policies than coding the fix.

I prefer being open to "free-style" contributions with automatic checks (as much as that is possible)

**Pull request merged**